mlflow/pipelines/steps/ingest/datasets.py

```diff
@@ -583,25 +583,36 @@ class SparkSqlDataset(_SparkDatasetMixin, _Dataset):
     (e.g. `SELECT * FROM my_spark_table`).
     """
 
-    def __init__(self, sql: str, dataset_format: str):
+    def __init__(self, sql: str, location: str, dataset_format: str):
         """
         :param sql: The Spark SQL query string that defines the dataset
                     (e.g. 'SELECT * FROM my_spark_table').
         :param dataset_format: The format of the dataset (e.g. 'csv', 'parquet', ...).
         """
         super().__init__(dataset_format=dataset_format)
         self.sql = sql
+        self.location = location
 
     def resolve_to_parquet(self, dst_path: str):
+        if self.location is None and self.sql is None:
+            raise MlflowException(
+                "Either location or sql configuration key must be specified for "
+                "dataset with format spark_sql"
+            ) from None
         spark_session = self._get_or_create_spark_session()
-        spark_df = spark_session.sql(self.sql)
+        spark_df = None
+        if self.location is not None:
+            spark_df = spark_session.table(self.location)
+        if self.sql is not None:
+            spark_df = spark_session.sql(self.sql)
```

Comment:
What happens when when both `self.location` and `self.sql` are specified? It could result in subtle bugs when the location table is different from the SQL query table. 

In addition, we should use `if-else if-else` clause to avoid reading the table twice.

Reply:
Yes! i forgot to add if..else. Good catch.

I think if `sql` is provided, we will choose that over the `location`. Location would be used to read the entire table if the sql is not provided as a fallback. 
What do you think?